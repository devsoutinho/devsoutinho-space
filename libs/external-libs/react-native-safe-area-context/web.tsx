export function useSafeAreaInsets() {
  return {
    top: 'env(safe-area-inset-top)',
    bottom: 'env(safe-area-inset-bottom)',
  }
}
