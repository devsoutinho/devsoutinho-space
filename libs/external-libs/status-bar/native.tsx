import { StatusBarProps } from './types';
import { StatusBar as Example } from 'expo-status-bar';

export function StatusBar({ style }: StatusBarProps) {
  return (
    <Example style={style} />
  );
}
