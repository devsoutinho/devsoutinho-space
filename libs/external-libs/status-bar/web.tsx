import { StatusBarProps } from './types';
import  Head from 'next/head';

export function StatusBar({ title }: StatusBarProps) {
  return (
    <Head>
      <title>{title}</title>
      <meta name="apple-mobile-web-app-status-bar-style" content="black" />
      {/* <meta name="theme-color" content="#EF4444" /> */}
    </Head>
  );
}
