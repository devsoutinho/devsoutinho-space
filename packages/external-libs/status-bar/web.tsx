import { StatusBarProps } from './types';
import  Head from 'next/head';

export function StatusBar({}: StatusBarProps) {
  return (
    <Head>
      <meta name="apple-mobile-web-app-status-bar-style" content="black" />
      {/* <meta name="theme-color" content="#EF4444" /> */}
    </Head>
  );
}
