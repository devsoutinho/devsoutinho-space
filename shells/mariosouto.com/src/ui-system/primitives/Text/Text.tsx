import BaseComponent from "@src/ui-system/theme/BaseComponent";
import { StyleSheet } from "@src/ui-system/theme/StyleSheet";
import React from "react";
import { typography } from "./typography";

export type TextVariant = keyof typeof typography;

interface TextProps {
  tag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span' | string;
  variant?: TextVariant;
  children?: React.ReactNode;
  styleSheet?: StyleSheet;
  onClick?: (event: any) => void;
  ref: any;
}
export const Text = React.forwardRef(({ tag, variant, styleSheet, children, ...props }: TextProps, ref) => {
  return (
    <BaseComponent
      as={tag}
      ref={ref}
      styleSheet={{
        ...typography[variant],
        ...styleSheet,
      }}
      {...props}
    >
      {children}
    </BaseComponent>
  )
})

Text.defaultProps = {
  tag: 'p',
  variant: 'body2',
}
