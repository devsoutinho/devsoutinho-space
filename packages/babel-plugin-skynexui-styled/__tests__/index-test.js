const plugin = require('../src/index');
const pluginTester = require('babel-plugin-tester').default;

const tests = [
  {
    title: 'import from "@skynexui/core (native)"',
    code: `import UIThemeProvider from '@skynexui/native/theme/provider/UIThemeProvider';
import Text from '@skynexui/native/components/Text';
import Box from '@skynexui/native/components/Box';`,
    snapshot: true,
    pluginOptions: { devNative: true }
  },
  {
    title: 'import from "@skynexui/core (web)"',
    code: `import UIThemeProvider from '@skynexui/native/theme/provider/UIThemeProvider';
import Text from '@skynexui/native/components/Text';
import Box from '@skynexui/native/components/Box';`,
    snapshot: true,
    pluginOptions: { devWeb: true }
  },
  {
    title: 'import from "native-native"',
    code: `import ReactNative from '@skynexui/native';
import Text from '@skynexui/native/components/Text';
import Box from '@skynexui/native/components/Box';`,
    snapshot: true
  },
  {
    title: 'import from "native-native"',
    code: `import ReactNative from '@skynexui/native';
import { View } from '@skynexui/native';
import { Invalid, View as MyView } from '@skynexui/native';
import * as ReactNativeModules from '@skynexui/native';`,
    snapshot: true,
  },
  {
    title: 'import from "@skynexui/web"',
    code: `import { unstable_createElement } from '@skynexui/web';
import { StyleSheet, View, TouchableOpacity, processColor } from '@skynexui/web';
import * as ReactNativeModules from '@skynexui/web';`,
    snapshot: true
  },
  {
    title: 'export from "@skynexui/native"',
    code: `export { View } from '@skynexui/native/components/Text';
export { StyleSheet, Text, unstable_createElement } from '@skynexui/native';`,
    snapshot: true
  },
  {
    title: 'export from "@skynexui/web"',
    code: `export { View } from '@skynexui/web';
export { StyleSheet, Text, unstable_createElement } from '@skynexui/web';`,
    snapshot: true
  },
  {
    title: 'require "@skynexui/native"',
    code: `const ReactNative = require('@skynexui/native/components/Box');
const Text = require('@skynexui/native/component/Text');
const { Bla, Sample } = require('@skynexui/native/component/Text');
const { StyleSheet, TouchableOpacity } = require('@skynexui/native');`,
    snapshot: true
  },
  {
    title: 'require "@skynexui/native"',
    code: `const ReactNative = require('@skynexui/native');
const { View } = require('@skynexui/native');
const { StyleSheet, TouchableOpacity } = require('@skynexui/native');`,
    snapshot: true,
  },
  {
    title: 'require "@skynexui/web"',
    code: `const ReactNative = require('@skynexui/web');
const { unstable_createElement } = require('@skynexui/web');
const { StyleSheet, View, TouchableOpacity, processColor } = require('@skynexui/web');`,
    snapshot: true
  }
];

pluginTester({
  babelOptions: {
    generatorOpts: {
      jsescOption: {
        quotes: 'single'
      }
    }
  },
  plugin,
  pluginName: 'Rewrite @skynexui/native to: ', // |core
  tests
});
