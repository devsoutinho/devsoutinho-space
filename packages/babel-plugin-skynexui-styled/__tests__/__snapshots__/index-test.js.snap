// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rewrite @skynexui/native to:  export from "@skynexui/native": export from "@skynexui/native" 1`] = `

export { View } from '@skynexui/native/components/Text';
export { StyleSheet, Text, unstable_createElement } from '@skynexui/native';

      ↓ ↓ ↓ ↓ ↓ ↓

export { View } from "@skynexui/web/components/Text";
export { StyleSheet } from "@skynexui/web";
export { Text } from "@skynexui/web";
export { unstable_createElement } from "@skynexui/web";


`;

exports[`Rewrite @skynexui/native to:  export from "@skynexui/web": export from "@skynexui/web" 1`] = `

export { View } from '@skynexui/web';
export { StyleSheet, Text, unstable_createElement } from '@skynexui/web';

      ↓ ↓ ↓ ↓ ↓ ↓

export { View } from "@skynexui/web";
export { StyleSheet, Text, unstable_createElement } from "@skynexui/web";


`;

exports[`Rewrite @skynexui/native to:  import from "@skynexui/core (native)": import from "@skynexui/core (native)" 1`] = `

import UIThemeProvider from '@skynexui/native/theme/provider/UIThemeProvider';
import Text from '@skynexui/native/components/Text';
import Box from '@skynexui/native/components/Box';

      ↓ ↓ ↓ ↓ ↓ ↓

import UIThemeProvider from "@skynexui/core/src/theme/provider/UIThemeProvider/native";
import Text from "@skynexui/core/src/components/Text/native";
import Box from "@skynexui/core/src/components/Box/native";


`;

exports[`Rewrite @skynexui/native to:  import from "@skynexui/core (web)": import from "@skynexui/core (web)" 1`] = `

import UIThemeProvider from '@skynexui/native/theme/provider/UIThemeProvider';
import Text from '@skynexui/native/components/Text';
import Box from '@skynexui/native/components/Box';

      ↓ ↓ ↓ ↓ ↓ ↓

import UIThemeProvider from "@skynexui/core/src/theme/provider/UIThemeProvider/web";
import Text from "@skynexui/core/src/components/Text/web";
import Box from "@skynexui/core/src/components/Box/web";


`;

exports[`Rewrite @skynexui/native to:  import from "@skynexui/web": import from "@skynexui/web" 1`] = `

import { unstable_createElement } from '@skynexui/web';
import { StyleSheet, View, TouchableOpacity, processColor } from '@skynexui/web';
import * as ReactNativeModules from '@skynexui/web';

      ↓ ↓ ↓ ↓ ↓ ↓

import { unstable_createElement } from "@skynexui/web";
import {
  StyleSheet,
  View,
  TouchableOpacity,
  processColor,
} from "@skynexui/web";
import * as ReactNativeModules from "@skynexui/web";


`;

exports[`Rewrite @skynexui/native to:  import from "native-native": import from "native-native" 1`] = `

import ReactNative from '@skynexui/native';
import Text from '@skynexui/native/components/Text';
import Box from '@skynexui/native/components/Box';

      ↓ ↓ ↓ ↓ ↓ ↓

import ReactNative from "@skynexui/web";
import Text from "@skynexui/web/components/Text";
import Box from "@skynexui/web/components/Box";


`;

exports[`Rewrite @skynexui/native to:  import from "native-native": import from "native-native" 2`] = `

import ReactNative from '@skynexui/native';
import { View } from '@skynexui/native';
import { Invalid, View as MyView } from '@skynexui/native';
import * as ReactNativeModules from '@skynexui/native';

      ↓ ↓ ↓ ↓ ↓ ↓

import ReactNative from "@skynexui/web";
import { View } from "@skynexui/web";
import { Invalid } from "@skynexui/web";
import { View as MyView } from "@skynexui/web";
import * as ReactNativeModules from "@skynexui/web";


`;

exports[`Rewrite @skynexui/native to:  require "@skynexui/native": require "@skynexui/native" 1`] = `

const ReactNative = require('@skynexui/native/components/Box');
const Text = require('@skynexui/native/component/Text');
const { Bla, Sample } = require('@skynexui/native/component/Text');
const { StyleSheet, TouchableOpacity } = require('@skynexui/native');

      ↓ ↓ ↓ ↓ ↓ ↓

const ReactNative = require("@skynexui/web/components/Box");

const Text = require("@skynexui/web/component/Text");

const Bla = require("@skynexui/web").default;

const Sample = require("@skynexui/web").default;

const StyleSheet = require("@skynexui/web").default;

const TouchableOpacity = require("@skynexui/web").default;


`;

exports[`Rewrite @skynexui/native to:  require "@skynexui/native": require "@skynexui/native" 2`] = `

const ReactNative = require('@skynexui/native');
const { View } = require('@skynexui/native');
const { StyleSheet, TouchableOpacity } = require('@skynexui/native');

      ↓ ↓ ↓ ↓ ↓ ↓

const ReactNative = require("@skynexui/web");

const View = require("@skynexui/web").default;

const StyleSheet = require("@skynexui/web").default;

const TouchableOpacity = require("@skynexui/web").default;


`;

exports[`Rewrite @skynexui/native to:  require "@skynexui/web": require "@skynexui/web" 1`] = `

const ReactNative = require('@skynexui/web');
const { unstable_createElement } = require('@skynexui/web');
const { StyleSheet, View, TouchableOpacity, processColor } = require('@skynexui/web');

      ↓ ↓ ↓ ↓ ↓ ↓

const ReactNative = require("@skynexui/web");

const { unstable_createElement } = require("@skynexui/web");

const {
  StyleSheet,
  View,
  TouchableOpacity,
  processColor,
} = require("@skynexui/web");


`;
